aea/context/base.py

```diff
@@ -31,6 +31,7 @@ class AgentContext:
 
     def __init__(self, agent_name: str,
                  public_keys: Dict[str, str],
+                 addresses: Dict[str, str],
```

Comment:
worth considering having a type like
```
Address = str
```

so we can then easily extend with other types if needed:
```
Address = Union[str, AnotherType]
```

Reply:
This comment applies to all the other places where `str` appears when referring to an address

Reply:
Agreed, though I'd prefer if we open a separate issue for that and deal with it another day. Not a priority atm.